lib/descriptor.js

```diff
@@ -24,3 +26,67 @@ const generateDescriptor = async (data, dpName = "scratchpad") => {
   return descriptor
 }
 module.exports.generateDescriptor = generateDescriptor
+
+const generateDescriptorForLocalFiles = function (dpName = "scratchpad") { 
+  return new Promise(function(resolve, reject) { 
```

Comment:
@zelima Since, there are several async functions, I could not get a final result. It returns descriptor without passing other functions like: fs.readFile, parse. That's why I used promise in order to complete  other async operations, and get final descriptor value. @anuveyatsu What do you think?